src/app/Http/Controllers/AdminController.php

```diff
@@ -1,10 +1,64 @@
 <?php
 
 namespace App\Http\Controllers;
+use App\Models\Order;
+use App\Models\Product;
+use App\Models\Categories;
+use App\Services\ManagerOrderService;
+use DB;
 
 class AdminController extends Controller
 {
-    public function index(){
-        return view('admin.index');
+    public function index(ManagerOrderService $managerOrderService)
+    {   
+        # get an array containing the number of orders for each month(lấy ra một mảng chứa số đơn đặt hàng của từng tháng).
+        $arr_Orders = Order::GetArrOrderMonth()->pluck('count_orders');
+        # get an array of months from 1 -> 12(lấy ra một mảng tháng từ 1 -> 12).
+        $arr_Months = Order::GetArrMonth()->pluck('month');
+        $number_of_orders = [0,0,0,0,0,0,0,0,0,0,0,0];
+
+        for ($i = 0; $i<count($arr_Months);$i++) {
+            $number_of_orders[$i] = $arr_Orders[$i];
+        }
+
+        # get top 5 most ordered products this month(lấy ra top 5 sản phẩm được đặt hàng nhiều nhất trong tháng này).
+        $datetime = $managerOrderService->getDatetime();
+        $amount_of_orders = Product::ProductJoinOrderProduct()
+        ->WhereProduct($datetime['this_month'])
+        ->SelectProducOrder()
+        ->groupBy('order_product.product_id')
+        ->orderBy('amount_of_order','desc')
+        ->limit(5)
+        ->pluck('amount_of_order');
+
+        $name_products = Product::ProductJoinOrderProduct()
+        ->WhereProduct($datetime['this_month'])
+        ->SelectProducName()
+        ->groupBy('order_product.product_id')
+        ->orderBy('amount_of_order','desc')
+        ->limit(5)
+        ->pluck('name');
+        
+        $amount_ofs = [0,0,0,0,0];
+        foreach($amount_of_orders as $index => $amount_of_order){
+            $amount_ofs[$index] = $amount_of_order;
+        }
+
+        $names = [0,0,0,0,0];
+        foreach($name_products as $index => $name_product){
+            $names[$index] = $name_product;
+        }
```

Comment:
Mục đích đoạn này để làm gì?

Reply:
cái này em thống kê top 5 sp được đặt hàng nhiều nhất trong 1 tháng:
- đầu tiên em cho 2 mảng $amount_ofs(sl đặt của sp đó trong 1 tháng) và $names(chứa 5 tên của sp đó) với dữ liệu mặc định là  0 cho 5 vị trí.sau đó em select và dùng vòng lặp for để lặp và add vô 2 mảng đó thôi anh.

Reply:
VD: Lần đầu chưa có người đặt. Chăng lẽ show ra là 0. name mà đặt 0 là thấy sai sai.
Sao không dùng array_push()?